packages/ui-plugin-sheets/src/controllers/contextmenu/contextmenu.controller.ts

```diff
@@ -58,7 +69,15 @@ export class SheetContextMenuController extends Disposable {
         const spreadsheetPointerDownObserver = objects.spreadsheet.onPointerDownObserver;
         const spreadsheetObserver = spreadsheetPointerDownObserver.add((event) => {
             if (event.button === 2) {
-                this._contextMenuService.triggerContextMenu(event, MenuPosition.CONTEXT_MENU);
+                const selections = this._selectionManagerService.getSelections();
+                const rangeType = selections?.length ? selections[0].range.rangeType : undefined;
+                if (rangeType === RANGE_TYPE.COLUMN) {
+                    this._contextMenuService.triggerContextMenu(event, SheetMenuPosition.COL_HEADER_CONTEXT_MENU);
```

Comment:
Could extract a function because there are similar pieces of code.

```ts
const triggerMenu = (position: SheetMenuPosition) => {
  this._contextMenuService.triggerContextMenu(event, position);
}
```